### PR TITLE
subctl must not display logger-related errors

### LINKED ISSUE #816: subctl must not display logger-related errors
When `subctl` ends up invoking code which logs, it produces logger-related errors:

```
subctl-0.16 deploy-broker --kubeconfig /home/vthapar/aws/acmhub/auth/kubeconfig 
 ✓ Setting up broker RBAC 
⢎⡠ Deploying the Submariner operator [controller-runtime] log.SetLogger(...) was never called, logs will not be displayed:
goroutine 1 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x65
sigs.k8s.io/controller-runtime/pkg/log.eventuallyFulfillRoot()
	sigs.k8s.io/controller-runtime@v0.15.0/pkg/log/log.go:59 +0xbd
sigs.k8s.io/controller-runtime/pkg/log.(*delegatingLogSink).Enabled(0xc000051080, 0xc00073c0d0?)
	sigs.k8s.io/controller-runtime@v0.15.0/pkg/log/deleg.go:111 +0x3d
github.com/go-logr/logr.Logger.Enabled(...)
	github.com/go-logr/logr@v1.2.4/logr.go:261
github.com/go-logr/logr.Logger.Info({{0x3750f98?, 0xc000051080?}, 0xc0004d11f8?}, {0xc0009bca00, 0x1f1}, {0x0, 0x0, 0x0})
	github.com/go-logr/logr@v1.2.4/logr.go:274 +0x78
github.com/submariner-io/admiral/pkg/log.Logger.Infof({{{0x3750f98?, 0xc000051080?}, 0x1de?}}, {0x32abaff?, 0x3283a5d?}, {0xc0004d11f8?, 0x2da2940?, 0x374dbf0?})
	github.com/submariner-io/admiral@v0.16.0-m1/pkg/log/logger.go:42 +0x74
github.com/submariner-io/admiral/pkg/util.maybeCreateOrUpdate.func1()
	github.com/submariner-io/admiral@v0.16.0-m1/pkg/util/create_or_update.go:148 +0xb05
k8s.io/client-go/util/retry.OnError.func1()
	k8s.io/client-go@v0.27.3/util/retry/util.go:51 +0x33
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x79feff?)
	k8s.io/apimachinery@v0.27.3/pkg/util/wait/wait.go:145 +0x4d
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0}, 0xc000a322c0?)
	k8s.io/apimachinery@v0.27.3/pkg/util/wait/backoff.go:461 +0x5f
k8s.io/client-go/util/retry.OnError({0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0}, 0x2b5f4a0?, 0xc000b45f10?)
	k8s.io/client-go@v0.27.3/util/retry/util.go:50 +0xa8
k8s.io/client-go/util/retry.RetryOnConflict(...)
	k8s.io/client-go@v0.27.3/util/retry/util.go:104
github.com/submariner-io/admiral/pkg/util.maybeCreateOrUpdate({0x374dbf0, 0xc00012e018}, {0x374efe0, 0xc00047b890}, {0x373a8c8?, 0xc000a322c0?}, 0xc0004d1478, 0x1)
	github.com/submariner-io/admiral@v0.16.0-m1/pkg/util/create_or_update.go:87 +0x1c8
github.com/submariner-io/admiral/pkg/util.CreateOrUpdate(...)
	github.com/submariner-io/admiral@v0.16.0-m1/pkg/util/create_or_update.go:67
github.com/submariner-io/subctl/pkg/resource.CreateOrUpdate({0x374dbf0?, 0xc00012e018?}, {0x374efe0?, 0xc00047b890?}, {0x373a8c8?, 0xc000a322c0?})
	github.com/submariner-io/subctl/pkg/resource/create_or_update.go:30 +0x74
github.com/submariner-io/subctl/pkg/clusterrolebinding.Ensure({0x374dbf0, 0xc00012e018}, {0x376a000?, 0xc000483040?}, 0x32b9b59?)
	github.com/submariner-io/subctl/pkg/clusterrolebinding/ensure.go:47 +0x59
github.com/submariner-io/subctl/pkg/clusterrolebinding.EnsureFromYAML({0x374dbf0, 0xc00012e018}, {0x376a000, 0xc000483040}, {0x32a773e, 0x13}, {0x3370d90, 0x114})
	github.com/submariner-io/subctl/pkg/clusterrolebinding/ensure.go:42 +0xc9
github.com/submariner-io/subctl/pkg/operator/ocp.EnsureRBAC({0x374dbf0, 0xc00012e018}, {0x371d5a0?, 0xc00086e270?}, {0x376a000, 0xc000483040}, {0x32a773e, 0x13}, {0xc0004d1638, 0x5, ...})
	github.com/submariner-io/subctl/pkg/operator/ocp/ensure.go:55 +0x172
github.com/submariner-io/subctl/pkg/submariner.Ensure({0x374dbf0, 0xc00012e018}, {0x3750530, 0xc000422be0}, {0x376a000, 0xc000483040}, {0x371d5a0, 0xc00086e270}, {0x32a773e, 0x13})
	github.com/submariner-io/subctl/pkg/submariner/ensure.go:69 +0x2b7
github.com/submariner-io/subctl/pkg/operator.Ensure({0x374dbf0, 0xc00012e018}, {0x3750530?, 0xc000422be0?}, {0x373f750, 0xc000781050}, {0x32a773e, 0x13}, {0xc0008f0a20, 0x30}, ...)
	github.com/submariner-io/subctl/pkg/operator/ensure.go:78 +0x4c5
github.com/submariner-io/subctl/pkg/deploy.deploy({0x374dbf0, 0xc00012e018}, 0x4c84f60, {0x3750530, 0xc000422be0}, {0x373f750?, 0xc000781050?})
	github.com/submariner-io/subctl/pkg/deploy/broker.go:99 +0x36f
github.com/submariner-io/subctl/pkg/deploy.Broker(0x4c84f60, {0x373f750, 0xc000781050}, {0x3750530, 0xc000422be0})
	github.com/submariner-io/subctl/pkg/deploy/broker.go:66 +0x22d
github.com/submariner-io/subctl/cmd/subctl.deployBrokerInContext(0xc000135b80, {0x32acf42, 0x15}, {0x3750530, 0xc000422be0})
	github.com/submariner-io/subctl/cmd/subctl/deploybroker.go:90 +0xa5
github.com/submariner-io/subctl/internal/restconfig.(*Producer).RunOnSelectedContext(0xc000336310, 0x33fe700, {0x3750530, 0xc000422be0})
	github.com/submariner-io/subctl/internal/restconfig/restconfig.go:230 +0x21d
github.com/submariner-io/subctl/cmd/subctl.glob..func5(0x4c52860?, {0x3284c7f?, 0x2?, 0x2?})
	github.com/submariner-io/subctl/cmd/subctl/deploybroker.go:54 +0x38
github.com/spf13/cobra.(*Command).execute(0x4c52860, {0xc0007829a0, 0x2, 0x2})
	github.com/spf13/cobra@v1.7.0/command.go:944 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0x4c4eee0)
	github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.7.0/command.go:992
github.com/submariner-io/subctl/cmd/subctl.Execute()
	github.com/submariner-io/subctl/cmd/subctl/root.go:67 +0x25
main.main()
	github.com/submariner-io/subctl/cmd/main.go:24 +0x17
 ✓ Deploying the Submariner operator 
 ✓ Created operator namespace: submariner-operator
 ✓ Created operator service account and role
 ✓ Created submariner service account and role
 ✓ Created lighthouse service account and role
 ✓ Deploying the broker 
 ✓ Saving broker info to file "broker-info.subm" 
```

`subctl` is supposed to track errors using its status framework. Instances such as the above should either be changed to only report errors without logging them (and have `subctl` handle the errors appropriately), or the logging framework should be set up to feed into the status framework (if that is possible). See https://github.com/submariner-io/admiral/blob/devel/pkg/log/kzerolog/kzerolog.go for the Kubernetes-based setup — this isn’t appropriate for `subctl`, but it shows what needs to be done.

### PR DESCRIPTION
**Pull request by tpantelis**
Fixes #816

### CODE REVIEW

**Review by skitt — APPROVED** [review]

File: cmd/subctl/root.go [hunk 1]
```diff
@@ -53,6 +55,8 @@ func init() {
 	runtime.Must(submarineropv1a1.AddToScheme(scheme.Scheme))
 
 	rest.SetDefaultWarningHandler(suppressWarnings{})
+
+	log.SetLogger(logr.New(log.NullLogSink{}))
```
Comment (vthapar) [inline cmd/subctl/root.go:60]: Why not use `log.SetLogger(logr.Discard())` ?

**Review by vthapar — CHANGES_REQUESTED** [review]

File: cmd/subctl/root.go (hunk 1 above)
Comment (tpantelis) [reply cmd/subctl/root.go:60]: Same effect but `Discard` is more readable.

Comment (tpantelis) [reply cmd/subctl/root.go:60]: Well `Discard` passes a nil `LogSink` which causes a nil pointer panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x21e16b0]

goroutine 1 [running]:
sigs.k8s.io/controller-runtime/pkg/log.(*loggerPromise).Fulfill(0xc000463c70, {0x0?, 0x0?})
	sigs.k8s.io/controller-runtime@v0.15.0/pkg/log/deleg.go:67 +0x30
sigs.k8s.io/controller-runtime/pkg/log.(*loggerPromise).Fulfill(0xc000463c20, {0x0?, 0x0?})
	sigs.k8s.io/controller-runtime@v0.15.0/pkg/log/deleg.go:83 +0x17d
sigs.k8s.io/controller-runtime/pkg/log.(*delegatingLogSink).Fulfill(...)
	sigs.k8s.io/controller-runtime@v0.15.0/pkg/log/deleg.go:192
sigs.k8s.io/controller-runtime/pkg/log.SetLogger({{0x0?, 0x0?}, 0x4c58fa0?})
	sigs.k8s.io/controller-runtime@v0.15.0/pkg/log/log.go:50 +0x4b
```

So I went back to using `NullLogSink`.

**Review by sridhargaddam — APPROVED** [review]

**Comment by sridhargaddam** [comment]
@vthapar can you re-review? This would fix the errors seen as part of testing 0.16.0-m2 images.

**Review by vthapar — APPROVED** [review]

**Pull request merged**